pkg/blobstore/local/closable_block_list.go

```diff
@@ -0,0 +1,81 @@
+package local
+
+import (
+	"github.com/buildbarn/bb-storage/pkg/blobstore/buffer"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// errClosedForWriting is an error code to indicate that the BlockList does not
+// accept any more write operations and ongoing write operations will fail.
+var errClosedForWriting = status.Error(codes.Unavailable, "Closed for writing")
+
+// ClosableBlockList is an implementation of BlockList that can be blocked for
+// writing. This is useful when shutting down to know that no more data is
+// written after the last sync.
+type ClosableBlockList struct {
+	BlockList
+
+	// Writes to closedForWriting must be synchronized with all write methods.
+	// This requirement is pushed down on the user of this struct.
+	closedForWriting bool
+}
```

Comment:
Could you maybe add a couple of lines here along the lines of:
```go
var _ BlockList = &ClosableBlockList{}
var _ PersistentStateSource = &ClosableBlockList{}
```
That documents which interfaces are implemented, and ensures that the build fails immediately when the contract is violated. This as opposed to causing a failure at the place of use.